### ISSUE TITLE
trying to load a schema i dumped with genson fails

### ISSUE THREAD

**Issue by ankopainting** [wolverdude/GenSON#14]
I'm just trying to reload a schema genSON created with add_schema()

```
  File "jason-voorhees.py", line 41, in load_schemas
    a.add_schema(json.loads(schema_body))
  File "/usr/local/lib/python2.7/site-packages/genson/generator.py", line 64, in add_schema
    self._add_properties(val, 'add_schema')
  File "/usr/local/lib/python2.7/site-packages/genson/generator.py", line 204, in _add_properties
    getattr(self._properties[prop], func)(val)
  File "/usr/local/lib/python2.7/site-packages/genson/generator.py", line 64, in add_schema
    self._add_properties(val, 'add_schema')
  File "/usr/local/lib/python2.7/site-packages/genson/generator.py", line 204, in _add_properties
    getattr(self._properties[prop], func)(val)
  File "/usr/local/lib/python2.7/site-packages/genson/generator.py", line 66, in add_schema
    self._add_items(val, 'add_schema')
  File "/usr/local/lib/python2.7/site-packages/genson/generator.py", line 208, in _add_items
    self._add_items_merge(items, func)
  File "/usr/local/lib/python2.7/site-packages/genson/generator.py", line 218, in _add_items_merge
    method(item)
  File "/usr/local/lib/python2.7/site-packages/genson/generator.py", line 58, in add_schema
    for prop, val in schema.items():
AttributeError: 'unicode' object has no attribute 'items'
```

**Comment by wolverdude** [comment]
Could you include the schema that you're adding here?

Also, what were the inputs you used to create the schema?

**Comment by regisoc** [comment]
Hi @wolverdude,

I have exactly the same problem. I will give you some details.

Base on this json file (remove the ".txt" in the extension): [es.json.txt](https://github.com/wolverdude/GenSON/files/756461/es.json.txt)

..I managed to generate very easily the associated schema (remove the ".txt" in the extension): [es.schema.txt](https://github.com/wolverdude/GenSON/files/756470/es.schema.txt) which is nice.

Then, I was trying to dump the schema in a JSON file, and reload it. 

I have tried someting like that with the example files I gave you above:

```python
import json
import genson

# paths
curr = os.path.abspath(".")
json_filepath = os.path.join(curr, "es.json")
schema_filepath = os.path.join(curr, "es.schema")

# build schema based on the given JSON file
json_data = None
with open(json_filepath) as f:
  json_data = json.loads(f.read())
s1 = genson.Schema()
s1.add_object(json_data)

# dump the schema in file
with open(schema_filepath, 'w') as f:
  json.dump(s1.to_dict(), f, separators=(',', ':'))

# prepare loading with another genson instance
s2 = genson.Schema()
json_schema_data = None
with open(schema_filepath, 'r') as f:
  json_schema_data = json.load(f)

# compare before adding/merging
print("same dict ? {}".format( s1.to_dict() == json_schema_data ))

# trying to load, aaaaand.... fail
s2.add_schema(json_schema_data) # fails
```

The trace:
```bash
same dict ? True

Traceback (most recent call last):
  File "main.py", line 129, in <module>
    s2.add_schema(json_schema_data)
  File "/home/nomad/workspace/test_jsonschema/lib/python3.5/site-packages/genson/generator.py", line 71, in add_schema
    self._add_properties(val, 'add_schema')
  File "/home/nomad/workspace/test_jsonschema/lib/python3.5/site-packages/genson/generator.py", line 258, in _add_properties
    getattr(self._properties[prop], func)(val)
  File "/home/nomad/workspace/test_jsonschema/lib/python3.5/site-packages/genson/generator.py", line 71, in add_schema
    self._add_properties(val, 'add_schema')
  File "/home/nomad/workspace/test_jsonschema/lib/python3.5/site-packages/genson/generator.py", line 258, in _add_properties
    getattr(self._properties[prop], func)(val)
  File "/home/nomad/workspace/test_jsonschema/lib/python3.5/site-packages/genson/generator.py", line 71, in add_schema
    self._add_properties(val, 'add_schema')
  File "/home/nomad/workspace/test_jsonschema/lib/python3.5/site-packages/genson/generator.py", line 258, in _add_properties
    getattr(self._properties[prop], func)(val)
  File "/home/nomad/workspace/test_jsonschema/lib/python3.5/site-packages/genson/generator.py", line 71, in add_schema
    self._add_properties(val, 'add_schema')
  File "/home/nomad/workspace/test_jsonschema/lib/python3.5/site-packages/genson/generator.py", line 258, in _add_properties
    getattr(self._properties[prop], func)(val)
  File "/home/nomad/workspace/test_jsonschema/lib/python3.5/site-packages/genson/generator.py", line 74, in add_schema
    self._add_items(val, 'add_schema')
  File "/home/nomad/workspace/test_jsonschema/lib/python3.5/site-packages/genson/generator.py", line 262, in _add_items
    self._add_items_merge(items, func)
  File "/home/nomad/workspace/test_jsonschema/lib/python3.5/site-packages/genson/generator.py", line 272, in _add_items_merge
    method(item)
  File "/home/nomad/workspace/test_jsonschema/lib/python3.5/site-packages/genson/generator.py", line 62, in add_schema
    for prop, val in schema.items():
AttributeError: 'str' object has no attribute 'items'
```

**Comment by wolverdude** [comment]
Thanks. After re-tracing the logic, I found the source again. The problem is [here](/wolverdude/GenSON/blob/1aa0245/genson/generator.py#L217). The schema builder is trying to be _too_ DRY in how it handles input. It's assuming that the `items` key will _always_ hold an array, which isn't actually true.

This goes back to a misunderstanding I had about how JSON Schema works when I originally created this library. I didn't properly understand the difference between [List and Tuple validation](https://spacetelescope.github.io/understanding-json-schema/reference/array.html) or how this distinction was stored in the schema.

I've fixed this in the `v1.0` branch, but that's still a work in progress. I need to figure out how to port that fix back to this branch. That said, even with the backport, `v0` still creates flawed schemas in this regard. `v1` will deprecate the whole `merge_arrays` option in favor of "training" the generator with a seed schema.
